test/unit/master/test_cluster_master.py

```diff
@@ -216,7 +216,10 @@ def test_handle_result_reported_from_slave_when_build_is_canceled(self):
         build_id = 1
         slave_url = "url"
         build = Build(BuildRequest({}))
+        self.patch('app.master.build.util')
+        build.generate_project_type()
```

Comment:
Ok. I ran this test without the added `build.generate_project_type()` call and it still passed, probably because the build is not in the `PREPARING` state. But if you remove the check for that state then this will probably be needed.